scripts/surface_high_engagement_issues.py

```diff
@@ -0,0 +1,503 @@
+        )
+        response.raise_for_status()
+        data = response.json()
+
+        items = data["data"]["organization"]["projectV2"]["items"]["nodes"]
+        all_items.extend(items)
+
+        page_info = data["data"]["organization"]["projectV2"]["items"]["pageInfo"]
+        has_next_page = page_info["hasNextPage"]
+        cursor = page_info["endCursor"]
+
+        # Print debug information
+        print(
+            f"Retrieved {len(items)} items, has_next_page: {has_next_page}, cursor: {cursor}"
+        )
+
+    project_items_cache = all_items
+    return all_items
+
+
+def get_project_item_id(issue_id: str, headers: dict) -> Union[str, None]:
+    """
+    Retrieve the project item ID for a given issue ID.
+
+    Args:
+        issue_id (str): The ID of the issue.
+        headers (dict): HTTP headers for GitHub API requests.
+
+    Returns:
+        str: The project item ID, or None if not found.
+    """
+    items = get_all_project_items(headers)
+    print(f"Total project items retrieved: {len(items)}")  # Debug statement
+
+    # Print all project item IDs and their content IDs
+    for item in items:
+        content_id = item["content"]["id"] if item["content"] else "None"
+        print(f"Project item ID: {item['id']}, Content ID: {content_id}")
+
+    for item in items:
+        if item["content"] and item["content"]["id"] == issue_id:
+            project_item_id = item["id"]
+            print(f"Matched project item ID: {project_item_id}")
+            return project_item_id
+
+    # Additional debugging: print a message if no match was found
+    print(f"No match found for issue ID {issue_id}")
+
+
+def get_single_select_options(project_id: str, headers: dict) -> list:
+    """
+    Retrieve the options for the single select field.
+
+    Args:
+        project_id (str): The ID of the project.
+        headers (dict): HTTP headers for GitHub API requests.
+
+    Returns:
+        list: List of options for the single select field.
+    """
+    global single_select_options_cache
+    if single_select_options_cache is not None:
+        "use da sso cache"
+        return single_select_options_cache
+
+    query = """
+    query($projectId: ID!) {
+      node(id: $projectId) {
+        ... on ProjectV2 {
+          fields(first: 100) {
+            nodes {
+              ... on ProjectV2SingleSelectField {
+                id
+                name
+                options {
+                  id
+                  name
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    """
+    variables = {"projectId": project_id}
+    url = "https://api.github.com/graphql"
+    response = httpx.post(
+        url, headers=headers, json={"query": query, "variables": variables}
+    )
+    response.raise_for_status()
+    data = response.json()
+
+    # Find the specific field by name and return its options
+    for field in data["data"]["node"]["fields"]["nodes"]:
```

Comment:
maybe we can we `assert` or `raise` here in case we don't have `data` in the response?

Reply:
ended up removing this due to refactor